### DISCUSSION TITLE
Hook custom task to CMake:build task

### DISCUSSION THREAD

**Discussion by dskulj1** [microsoft/vscode-cmake-tools#2785] [Q&A] [answered]
Is it possible to create a custom task in the task.json file and hook that task to the original CMake:Build task, so they both are run on the Build button from the CMake toolbar?

**Comment by elahehrashedi** [comment, marked as answer]
use this setting: `"cmake.buildTask": true` to use tasks instead of the normal build process. Using this setting, when you hit the build button, you will be asked to choose a task to run. In case you want to bypass the step in which you have to choose the build task, you can define your build task as the `default` task.
```
		{
			"type": "cmake",
			"label": "CMake: build",
			"command": "build",
			"targets": [
				"myTarget"
			],
			"group":  {
				"isDefault": true,
				"kind": "build"
			},
			"problemMatcher": [],
			"detail": "CMake template build task"
		},

```

**Reply by elahehrashedi** [reply]
And if you want to run two tasks, you can make your task depend on another task by adding this to the task definition: `"dependsOn": ["The name of the task to run before build task"]`

**Reply by dskulj1** [reply]
Thank you very much, it helps a lot. But, I realized that I needed to override CMake:Configure task, since I have to set up the environment before the configure is executed. First, I tried to override CMake:Configure and just hit the build button, it didn't work. Then I tried this, but it still didn't work.

"tasks": [
		{
			"type": "cmake",
			"label": "CMake: build",
			"command": "build",
			"targets": [
				"all"
			],
			"group": {
				"kind": "build",
				"isDefault": true
			},
			"problemMatcher": [],
			"detail": "CMake template build task",
			"dependsOn":["CMake: configure"]
		},
		{
			"label": "vsvars_env",
			"type": "shell",
			"windows": {
				"command": "call \"C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Auxiliary\\Build\\vcvarsall.bat\"",
				"args": ["amd64_x86", "-vcvars_ver=14.29 10.0.19041.0"]
			}
		},
		{
			"type": "cmake",
			"label": "CMake: configure",
			"command": "configure",
			"problemMatcher": [],
			"group": {
				"isDefault": true
			},
			"detail": "CMake template configure task",
			"dependsOn": [ "vsvars_env"]
		}
	]

Are there any Cmake settings that should be set like for CMake:Build command?
